HACKING.rst

```diff
@@ -423,6 +423,143 @@ This can be done incrementally, one function at a time:
 * finally, remove it (and any other now-unused functions) from
   cloudinit.net (to avoid having two parallel implementations)
 
+``cloudinit.net`` Functions/Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The functions/classes that need refactoring break down into some broad
+categories:
+
+* helpers for accessing ``/sys`` (that should not be on the top-level
+  ``Networking`` class as they are Linux-specific):
+
+  * ``get_sys_class_path``
+  * ``sys_dev_path``
+  * ``read_sys_net``
+  * ``read_sys_net_safe``
+  * ``read_sys_net_int``
+
+* those that directly access ``/sys`` (via helpers) and should (IMO) be
+  included in the API of the ``Networking`` class:
+
+  * ``generate_fallback_config``
+  * ``get_ib_interface_hwaddr``
+  * ``get_interface_mac``
+  * ``interface_has_own_mac``
+  * ``is_bond``
+  * ``is_bridge``
+  * ``is_connected``
+  * ``is_physical``
+  * ``is_present``
+  * ``is_renamed``
+  * ``is_up``
+  * ``is_vlan``
+  * ``is_wireless``
+  * ``wait_for_physdevs``
+
+* those that directly access ``/sys`` (via helpers) but may be
+  Linux-specific concepts or names:
+
+  * ``get_master``
+  * ``device_devid``
+  * ``device_driver``
```

Comment:
most other Unices have the name of the device driver in the name of the device

Reply:
So it sounds like a function which extracts that driver from the device name would still make sense?

(For my own information: what do such device names look like?)

Reply:
Do we need to expose that in a public API? I've the feeling the information is just required by the backend driver itself.

Reply:
> Do we need to expose that in a public API?

This isn't a public API, it is only intended for internal cloud-init use.

> I've the feeling the information is just required by the backend driver itself.

`device_driver` is called by a bunch of Linux-specific functions, but also `extract_physdevs` and `generate_fallback_config`. These latter two are expected to work on every platform, so I don't think we can only have `device_driver` present on `LinuxNetworking` in this initial layout. It's possible that as we perform the refactor we will discover that we don't need `device_driver` on the super-class after all, and I think we can move it at that point.

Does that sound right (and reasonable)?